service/worker/handler/consistency_check_handlers.go

```diff
@@ -17,259 +17,403 @@ package handler
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
+	"iter"
 	"strings"
-	"time"
 
 	"github.com/hibiken/asynq"
-	"github.com/minio/minio-go/v7"
-	"github.com/rs/zerolog"
 
 	"github.com/clyso/chorus/pkg/dom"
+	"github.com/clyso/chorus/pkg/entity"
 	"github.com/clyso/chorus/pkg/log"
-	"github.com/clyso/chorus/pkg/storage"
+	"github.com/clyso/chorus/pkg/rclone"
+	"github.com/clyso/chorus/pkg/store"
 	"github.com/clyso/chorus/pkg/tasks"
-	"github.com/clyso/chorus/pkg/util"
 )
 
 const (
 	CEmptyDirETagPlaceholder = "d"
 )
 
-func (s *svc) HandleConsistencyCheck(ctx context.Context, t *asynq.Task) (err error) {
+type ConsistencyCheckCtrl struct {
+	svc      *ConsistencyCheckSvc
+	queueSvc tasks.QueueService
+}
+
+func NewConsistencyCheckCtrl(svc *ConsistencyCheckSvc, queueSvc tasks.QueueService) *ConsistencyCheckCtrl {
+	return &ConsistencyCheckCtrl{
+		svc:      svc,
+		queueSvc: queueSvc,
+	}
+}
+
+func (r *ConsistencyCheckCtrl) HandleConsistencyCheck(ctx context.Context, t *asynq.Task) error {
 	var payload tasks.ConsistencyCheckPayload
 	if err := json.Unmarshal(t.Payload(), &payload); err != nil {
 		return fmt.Errorf("unable to unmarshal paylaod: %w", err)
 	}
 
 	locationCount := len(payload.Locations)
-	storages := make([]string, 0, locationCount)
-
 	if locationCount == 0 {
 		return fmt.Errorf("migration location list is empty: %w", asynq.SkipRetry)
 	}
 
-	if err := s.storageSvc.StoreConsistencyCheckID(ctx, payload.ID); err != nil {
-		return fmt.Errorf("unable to record consistency check id: %w", err)
+	locations := make([]entity.ConsistencyCheckLocation, 0, locationCount)
+	for _, payloadLocation := range payload.Locations {
+		locations = append(locations, entity.NewConsistencyCheckLocation(payloadLocation.Storage, payloadLocation.Bucket))
 	}
 
-	for _, location := range payload.Locations {
-		listTask := tasks.ConsistencyCheckListPayload{
-			MigrateLocation: location,
-			StorageCount:    uint8(locationCount),
-			ID:              payload.ID,
-		}
-
-		if err := s.storageSvc.IncrementConsistencyCheckScheduledCounter(ctx, payload.ID, 1); err != nil {
-			return fmt.Errorf("unable to increment consistency check scheduled counter: %w", err)
-		}
+	shouldCheckVersions, err := r.svc.ShouldCheckVersions(ctx, payload.User, locations)
```

Comment:
is it makes sense to check bucket type in grpc handler, or even make it as api request parameter?  This will allow user to check only latest versions for versioned bucket and in we can also remove `IsBuckedVersioned` method from rclone service interface.

Reply:
> is it makes sense to check bucket type in grpc handler

As I see, it is more logical to do it here, in the basic task. It splits scans into storage streams, therefore from my pov it is the best place to do it so far.

> make it as api request paramete

It can be a part of check parameters and further improvements, alongside with etag and size validation skips.

> we can also remove IsBuckedVersioned method from rclone service interface.

Doubt we can remove it. We'd still need to check it, if e.g. user is not proivding any parameters.